web/app.js

```diff
@@ -2331,8 +2333,21 @@ function webViewerUpdateViewarea({ location }) {
   const currentPage = PDFViewerApplication.pdfViewer.getPageView(
     /* index = */ PDFViewerApplication.page - 1
   );
-  const loading = currentPage?.renderingState !== RenderingStates.FINISHED;
-  PDFViewerApplication.toolbar?.updateLoadingIndicatorState(loading);
+
+  if (PDFViewerApplication._loadIconTimeoutId !== null) {
+    clearTimeout(PDFViewerApplication._loadIconTimeoutId);
+    PDFViewerApplication._loadIconTimeoutId = null;
+  }
+
+  if (currentPage?.renderingState !== RenderingStates.FINISHED) {
+    PDFViewerApplication._loadIconTimeoutId = setTimeout(() => {
+      PDFViewerApplication._loadIconTimeoutId = null;
+      const loading = currentPage?.renderingState !== RenderingStates.FINISHED;
```

Comment:
Also, this isn't even correct since there's nothing that says that `currentPage` is actually still *the current page* when the timeout resolves.